openedx_external_enrollments/external_enrollments.py

```diff
@@ -0,0 +1,424 @@
+                openedx_user, openedx_profile = get_user(email=email)
+                # TODO do not force logic assuming names with 2 words
+                first_name, last_name = openedx_profile.name.split(" ", 1)
+                user["FirstName"] = first_name.strip(" ")
+                user["LastName"] = last_name.strip(" ")
+                user["Email"] = email
+            except Exception:
+                pass
+
+        return user
+
+    def _get_salesforce_data(self, data):
+        """
+
+        :param data:
+        :return:
+        """
+        salesforce_data = {}
+        order_lines = data.get("supported_lines")
+        if len(order_lines) > 0:
+            try:
+                salesforce_data.update(self._get_program_of_interest_data(data, order_lines))
+                salesforce_data["Purchase_Type"] = "Program" if data.get("program") else "Course"
+                salesforce_data["PaymentAmount"] = data.get("paid_amount")
+                salesforce_data["Amount_Currency"] = data.get("currency")
+            except Exception:
+                pass
+
+        return salesforce_data
+
+    def _get_program_of_interest_data(self, data, order_lines):
+        """
+
+        :param data:
+        :param order_lines:
+        :return:
+        """
+        program_of_interest = {}
+        program = data.get("program")
+        try:
+            email = order_lines[0].get("user_email")
+            openedx_user, _ = get_user(email=email)
+            request_time = datetime.datetime.utcnow()
+            if program:
+                related_program = ProgramSalesforceEnrollment.objects.get(bundle_id=program.get("uuid"))
+                program_of_interest = related_program.meta
+                program_of_interest["Drupal_ID"] = "enrollment+program+{}+{}".format(
+                    openedx_user.username,
+                    request_time.strftime("%Y/%m/%d-%H:%M:%S"),
+                )
+            else:
+                single_course = order_lines[0]
+                course = self._get_course(single_course.get("course_id"))
+                program_of_interest = course.other_course_settings.get("salesforce_data")
+                program_of_interest["Drupal_ID"] = "enrollment+course+{}+{}".format(
+                    openedx_user.username,
+                    request_time.strftime("%Y-%m-%d-%H:%M:%S"),
+                )
+
+            program_of_interest["Lead_Source"] = data.get(
+                "utm_source",
+                program_of_interest.get("Lead_Source", "Undefined"),
+            )
+            program_of_interest["Secondary_Source"] = data.get(
+                "utm_campaign",
+                program_of_interest.get("Secondary_Source", "Undefined"),
+            )
+            program_of_interest["Tertiary_Source"] = data.get(
+                "utm_medium",
+                program_of_interest.get("Tertiary_Source", "Undefined"),
+            )
+        except:
```

Comment:
Some kind of logging info? @andrey-canon

Reply:
out of the scope of this pr